dev-lang/swift_lang/swift_lang-3.1.recipe

```diff
@@ -0,0 +1,158 @@
+
+PROVIDES="
+	swift_lang$secondaryArchSuffix = $portVersion compat >= 1.2
+	cmd:swift$secondaryArchSuffix = $portVersion
+	cmd:swiftc$secondaryArchSuffix = $portVersion
+	cmd:swift_api_digester$secondaryArchSuffix = $portVersion
+	cmd:swift_autolink_extract$secondaryArchSuffix = $portVersion
+	cmd:swift_demangle$secondaryArchSuffix = $portVersion
+	cmd:swift_format$secondaryArchSuffix = $portVersion
+	cmd:swift_ide_test$secondaryArchSuffix = $portVersion
+	cmd:swift_llvm_opt$secondaryArchSuffix = $portVersion
+	cmd:swift_reflection_dump$secondaryArchSuffix = $portVersion
+	cmd:swift_remoteast_test$secondaryArchSuffix = $portVersion
+	cmd:sil_func_extractor$secondaryArchSuffix = $portVersion
+	cmd:sil_opt$secondaryArchSuffix = $portVersion
+	cmd:sil_nm$secondaryArchSuffix = $portVersion
+	cmd:sil_passpipeline_dumper$secondaryArchSuffix = $portVersion
+	cmd:lldb_moduleimport_test$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	llvm${secondaryArchSuffix} >= 3.7.0 base
+	lib:libuuid$secondaryArchSuffix
+	lib:libicuuc$secondaryArchSuffix
+	lib:libicui18n$secondaryArchSuffix
+	lib:libicudata$secondaryArchSuffix
+	lib:libedit$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	lib:libxml2$secondaryArchSuffix
+	lib:libexecinfo$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libcharset$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libsqlite3$secondaryArchSuffix
+	"
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libuuid$secondaryArchSuffix
+	devel:libicuuc$secondaryArchSuffix
+	devel:libicui18n$secondaryArchSuffix
+	devel:libicudata$secondaryArchSuffix
+	devel:libexecinfo$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libcharset$secondaryArchSuffix
+	devel:libedit$secondaryArchSuffix
+	devel:libsqlite3$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	devel:libxml2$secondaryArchSuffix
+	gcc_syslibs$secondaryArchSuffix_devel
+	devel:libgcc$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	llvm$secondaryArchSuffix_clang
+	gcc_syslibs$secondaryArchSuffix
+	gcc_syslibs_devel$secondaryArchSuffix
+	devel:libgcc$secondaryArchSuffix
+	cmd:libtool$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	cmd:clang$secondaryArchSuffix
+	cmd:clang++$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:swig
+	cmd:make
+	cmd:python
+	cmd:cmake
+	cmd:find
+	cmd:diff
+	cmd:sed
+	cmd:grep
+	cmd:groff
+	cmd:python3
+	cmd:which
+	"
+
+BUILD()
+{
+	cd ..
+
+	mkdir -p swift
+	mkdir -p clang
+	mkdir -p llvm
+	mkdir -p cmark
+
+	# Copy all the source files
+
+	cp -rd $sourceDir/* swift
+	cp -rd $sourceDir2/* clang
+	cp -rd $sourceDir3/* llvm
+	cp -rd $sourceDir4/* cmark
+
+	# Fix permisssions for build-script.
+	chmod +x ./swift/utils/build-script
+	chmod +x ./swift/utils/build-script-impl
+	chmod -R 755 ./swift/utils
+
+	export HOST_CC="/bin/clang"
+	export HOST_CXX="/bin/clang++"
+	export CXXFLAGS="-fPIC"
+	export CFLAGS="-fPIC"
```

Comment:
Building the toolchain without the PIE flag isn't possible. It throws symbol relocation errors whilst linking some object files. This [gist](https://gist.github.com/return/d04471f208873d8a07b005ebb6290e4b) outline what happens when building without this flag.